src/prpy/tsr/tsr.py

```diff
@@ -186,9 +189,12 @@ def sample_xyzrpy(self, xyzrpy=NANBW):
         if not all(check):
             raise ValueError('xyzrpy must be within bounds', check)
 
-        return [self.Bw[i, 0] + (self.Bw[i, 1] - self.Bw[i, 0]) *
-                numpy.random.random_sample()
-                if numpy.isnan(x) else x for i, x in enumerate(xyzrpy)]
+        Bw_sample = [self._Bw_cont[i, 0] + (self._Bw_cont[i, 1] - self._Bw_cont[i, 0]) *
+                     numpy.random.random_sample()
+                     if numpy.isnan(x) else x for i, x in enumerate(xyzrpy)]
+        # Unwrap rpy to [-pi, pi]		
+        Bw_sample[3:6] = numpy.add(Bw_sample[3:6], pi) % (2*pi) - pi
```

Comment:
I believe this was done by @rachelholladay. I'll track it down and figure it out.